test/src/code_field/code_controller_tab_test.dart

```diff
@@ -73,5 +95,70 @@ public class MyClass {
         );
       },
     );
+
+    test('works with several tab insertion', () {
+      final controller = CodeController(
+        text: snippetWithTabs,
+        language: java,
+        modifiers: [const TabModifier()],
```

Comment:
Please also make `TabModifier` do nothing because we no longer use its logic for replacement.